python/sglang/srt/entrypoints/openai/usage_processor.py

```diff
@@ -0,0 +1,69 @@
+from __future__ import annotations
+
+from typing import Any, Dict, List, Mapping, Optional
+
+from sglang.srt.entrypoints.openai.utils import calculate_token_usage
+
+
+def _details_if_cached(count: int) -> Optional[Dict[str, int]]:
+    """Return {"cached_tokens": N} only when N > 0 (keeps JSON slim)."""
+    return {"cached_tokens": count} if count > 0 else None
+
+
+class UsageProcessor:
+    """Stateless helpers that turn raw token counts into a UsageInfo."""
+
+    @classmethod
+    def calculate_response_usage(
+        cls,
+        responses: List[Dict[str, Any]],
+        *,
+        n_choices: int = 1,
+        enable_cache_report: bool = False,
+    ):
+        completion = sum(r["meta_info"]["completion_tokens"] for r in responses)
```

Comment:
Same, I feel the original naming is more straight-forward.

```suggestion
        # Sum completion tokens from all responses
    completion_tokens = sum(
        response["meta_info"]["completion_tokens"] for response in responses
    )

    # For prompt tokens, only count every n_choices-th response to avoid double counting
    prompt_tokens = sum(
        responses[i]["meta_info"]["prompt_tokens"]
        for i in range(0, len(responses), n_choices)
    )
```